pkg/api/plugin.go

```diff
@@ -41,4 +47,6 @@ type Plugin interface {
 	HealthCheck(ctx context.Context, cs *OpenShiftManagedCluster) error
 
 	Update(ctx context.Context, cs *OpenShiftManagedCluster, azuredeploy []byte) error
+
+	GetPluginConfig() PluginConfig
```

Comment:
I don't think exposing this in the plugin interface is necessary. Both `Deploy` and `Update` can accept `PluginConfig` via an argument provided by our fake RP implementation.

Reply:
@kargakis the problem is that we call Initialzer with a plugin instance in createorupdate. So if I remove this api I get the following error:

# github.com/openshift/openshift-azure/cmd/createorupdate
cmd/createorupdate/createorupdate.go:117:23: cannot use p (type "github.com/openshift/openshift-azure/pkg/api".Plugin) as type initialize.Initializer in argument to upgrade.Deploy:
	"github.com/openshift/openshift-azure/pkg/api".Plugin does not implement initialize.Initializer (missing GetPluginConfig method)
make: *** [Makefile:5: build] Error 2

Reply:
don't worry, figured out what you meant now.. fixing..